packages-next/fields-document/src/DocumentEditor/Toolbar.tsx

```diff
@@ -0,0 +1,237 @@
+/** @jsx jsx */
+
+import { jsx } from '@keystone-ui/core';
+
+import { ReactEditor, useSlate } from 'slate-react';
+
+import { Fragment, ReactNode, useState } from 'react';
+
+import { Editor, Transforms } from 'slate';
+
+import { Button, Spacer } from './components';
+import { LinkButton } from './link';
+import { insertPanel } from './panel';
+import { insertQuote } from './quote';
+import { BlockComponentsButtons } from './component-blocks';
+import { Mark, isMarkActive, onlyContainerNodeInCurrentSelection, toggleMark } from './utils';
+import { Hoverable } from './components/hoverable';
+import { insertColumns } from './columns';
+import { ListButton } from './lists';
+import { insertBlockquote } from './blockquote';
+
+function getCanDoAlignment(editor: ReactEditor) {
+  const [node] = Editor.nodes(editor, {
+    match: node => node.type === 'paragraph',
+  });
+  return !!node;
+}
+
+// TODO use icons for toolbar buttons, make it sticky, etc
+export const Toolbar = () => {
+  const editor = useSlate();
+  const shouldInsertBlock = onlyContainerNodeInCurrentSelection(editor);
+  const canDoAlignment = getCanDoAlignment(editor);
+  return (
+    <ToolbarContainer>
+      <Headings />
+      <ListButton type="unordered-list">• List</ListButton>
+      <ListButton type="ordered-list"># List</ListButton>
+      <Spacer />
+      <InlineTextThings />
+      <Spacer />
+      <LinkButton />
+      <BlockComponentsButtons shouldInsertBlock={shouldInsertBlock} />
+      <Button
+        onMouseDown={event => {
+          event.preventDefault();
+          insertColumns(editor);
+        }}
+      >
+        + Columns
+      </Button>
+      {/*
```

Comment:
AccessBoundary has been deleted for now, right? Shouldn't we delete this code as well (not just comment it out)?